main.go

```diff
@@ -133,8 +176,17 @@ func (c *Collector) Collect(ch chan<- prometheus.Metric) {
 			return
 		}
 
-		for _, resource := range filteredResources {
-			c.collectResource(ch, resource, metricsStr, resourceGroup.Aggregations)
+		for _, f := range filteredResources {
+			var resource resourceMeta
+			resource.metrics = metricsStr
+
+			if len(resourceGroup.Aggregations) > 0 {
```

Comment:
Could we factorize this ... it seems the same lines of code as for all kind of filter